core/src/main/java/de/arbeitsagentur/opdt/keycloak/cassandra/connection/DefaultCassandraConnectionProviderFactory.java

```diff
@@ -164,9 +172,20 @@ public void init(Config.Scope scope) {
         repository = createRepository(cqlSession);
     }
 
+  public CqlSessionBuilder configure(CqlSessionBuilder cqlSessionBuilder) {
```

Comment:
Thanks @opdt I was having problems getting the migration to run in the tests without a timeout. Can you point me to some documentation/examples of how this is achieved using external file based configuration? I'm not super familiar with cassandra, so the options are new to me.

Reply:
I've been struggling to figure out how to load an external file based configuration @opdt. Is there an example you're aware of?

Reply:
Finally figured out how to load config from a file. I have a branch here, so let me know if you want me to do a separate PR. https://github.com/opdt/keycloak-cassandra-extension/compare/main...xgp:keycloak-cassandra-extension:xgp/config-file?expand=1

Reply:
You dont need any custom code to load the config from a file. The cassandra driver does it automatically: https://docs.datastax.com/en/developer/java-driver/4.17/manual/core/configuration/ (application.conf in Classpath)

Reply:
Correct. But if you want to allow it to be put somewhere else and **not** bundled with the jar, then you have to tell Cassandra where to load it from.

Reply:
Thats possible via `-Dconfig.file=/path/to/application.conf`via the underyling Lightbend configuration library (which also supports loading from other resources/URLs).

Reply:
The problem I encounter when I specify that is that the [init method](https://github.com/opdt/keycloak-cassandra-extension/blob/main/core/src/main/java/de/arbeitsagentur/opdt/keycloak/cassandra/connection/DefaultCassandraConnectionProviderFactory.java#L125) still runs. So even though the underlying library supports the `-Dconfig.file` override, it appears that it never gets there because of the exceptions thrown when the variables are missing. 

Also, if I set both, the variables in the config scope are what gets used.